lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -145,6 +145,68 @@ class MatmulOpConversionPattern : public OpConversionPattern<ttir::MatmulOp> {
 };
 // ANCHOR_END: adding_an_op_matmul_op_rewriter
 
+class Conv2dOpConversionPattern : public OpConversionPattern<ttir::Conv2dOp> {
+public:
+  using OpConversionPattern<ttir::Conv2dOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(ttir::Conv2dOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    // auto kernel_ty = mlir::cast<RankedTensorType>(adaptor.getWeight().getType());
+    // llvm::ArrayRef<std::int64_t> kernel_shape = kernel_ty.getShape();
```

Comment:
don't forget to remove commented code unless there's a strong reason to leave it behind